### ISSUE TITLE
Failed to start workspaces using devfile spec v2.1.0

### ISSUE THREAD

**Issue by l0rd** [eclipse-che/che#19299]
### Describe the bug

<img width="700" alt="image" src="https://user-images.githubusercontent.com/606959/111306596-b103ea80-8658-11eb-8fb6-d0ff44e8b5e0.png">

When I try to start a workspace using a factory link it fails with "Failed to resolve a devfile."

### Che version

Nightly

### Steps to reproduce

Deploy Che, on OpenShift, with devworkspace support enabled

```bash
$ cat patch.yaml
spec:
  devWorkspace:
    enable: true
$ chectl server:deploy --che-operator-cr-patch-yaml=patch.yaml \
                     -p openshift --no-olm-suggested-namespace \
                     -n workspaces-server
```

Open a factory link

```bash
open ${CHE_URL}/#https://github.com/l0rd/spring-petclinic/                     # <== devfile v2 
open ${CHE_URL}/#https://github.com/l0rd/spring-petclinic/tree/che-happy-path  # <== devfile v1
```

### Runtime

OpenShift 4.7.1

### Installation method

```bash
$ chectl version
chectl/0.0.20210315-next.19bcb80 darwin-x64 node-v12.21.0
```

### Environment

OpenShift dev cluster

### More screenshots

<img width="1531" alt="image" src="https://user-images.githubusercontent.com/606959/111311088-13131e80-865e-11eb-9d30-60a102413806.png">

**Comment by sleshchenko** [comment]
With the nightly Che I have error displayed on Dashboard, and it's the error that Che is not able to find a valid Devfile v1 or v2.
![Screenshot_20210316_153353](https://user-images.githubusercontent.com/5887312/111317858-3396a500-866d-11eb-8727-42cdae692aeb.png)

When I try direct link to devfile-v2.yaml (/#https://raw.githubusercontent.com/l0rd/spring-petclinic/master/devfile-v2.yaml) I have another error:
![Screenshot_20210316_153602](https://user-images.githubusercontent.com/5887312/111318081-6476da00-866d-11eb-86f2-97ee5d7841a5.png)

**Comment by skabashnyuk** [comment]
AFAIK fixed with https://github.com/eclipse-che/che-machine-exec/pull/139

**Comment by skabashnyuk** [comment]
We only support `1.0.0` and `2.0.0`. @l0rd  are you sure that `2.1.0` is a valid version?

**Comment by l0rd** [comment]
The version we should target is `2.1.0`

**Comment by l0rd** [comment]
To be clear, [2.1.0 hasn't been released yet](https://github.com/devfile/api/releases), but that's the devfile/api release that is supposed to include everything required by Che (i.e. 2.0.0 is missing some stuff like cpu limits that are required by Che).

**Comment by skabashnyuk** [comment]
@l0rd do you want to enable 2.1.0 after official release?

**Comment by l0rd** [comment]
@skabashnyuk I was expecting that we already supported `2.1.0` (`2.1.0-alpha` to be precise). In particular because if I am not wrong the DevWorkspace operator that we deploy with Che supports that version of the API, not `2.0.0`.

**Comment by skabashnyuk** [comment]
> I was expecting that we already supported 2.1.0

We don't support that version at this moment. 
To avoid misunderstanding,  can you create an issue explaining what and in which form/version we should support?

**Comment by l0rd** [comment]
> To avoid misunderstanding, can you create an issue explaining what and in which form/version we should support?

You tell me. I would like to support latest version of the spec. `DevWorkspace` operator supports v `2.1.0-alpha` (@sleshchenko keep me honest). Hence it make sense for che-server to be consistent with that.

On the other hand, if aligning Che server and DevWorkspace is hard we can wait the release of `2.1.0` that we should be part of milestone 2.

@sleshchenko @amisevsk @JPinkney what do you think?

**Comment by amisevsk** [comment]
We've thus far been rolling with whatever the latest schemaVersion for devfile 2.x is. Since 2.x is supposed to be backwards compatible, our check is basically the regex `2\..*`. 

However, on the Go side, trying to deserialize unknown fields isn't an error -- if DWO supports only 2.0 (no cpuLimit) they'll just be ignored if you throw a 2.1.0 devfile at it. Since we don't have a full 2.1.0 release yet, currently DWO supports devfile features as of commit https://github.com/devfile/api/commit/283b0c54946e9fea9872c25e1e086c303688f0e8.

**Comment by sleshchenko** [comment]
> our check is basically the regex 2\..*.

to be honest, that check is done only when we fetch Devfile via ID or URI(devfile plugin component), but the goal is to rework it to DevWorkspaceTemplate, or both.

DevWorkspace CRD itself does not have `schemaVersion` at all(since it's devfile metadata which is not part of devfile which is included into devworkspace.spec.template).
DevWorkspace CRD has only the following versions:
- v1alpha1(targets devfile v2.0.0 but when it was named [v1alpha1](https://github.com/devfile/api/tree/v1alpha1));
- v1alpha2(targets some state of devfile v2.1.0, as Angel mentioned, the current state is https://github.com/devfile/api/commit/283b0c54946e9fea9872c25e1e086c303688f0e8).

Go side just ignore unknown fields, but K8s API (or it's kubectl I'm not sure) does validation and throw an error when unknown field is used, and tell how to skip validation.

**Comment by sleshchenko** [comment]
A dedicated issue is created to make Che Server embeds v2.1.0-alpha JSON Schema https://github.com/eclipse/che/issues/19304.

What is plan with 2.1.0-alpha? Is it going to be dropped after v2.1.0 is released?
And then users will have to update their devfiles? Or v2.1.0-alpha will live forever (at least at some git tag) and will be aligned with released 2.1.0?

**Comment by l0rd** [comment]
Closing this issue. Opening a new one https://github.com/eclipse/che/issues/19322
